server/internal/gateway/gateway.go

```diff
@@ -311,6 +311,18 @@ func New(glue glue.Glue) (glue.Gateway, error) {
 		p.Go(p.worker)
 	}
 
+	// monitor channel length
+	go p.monitorChannelLen()
+
 	isOk = true
 	return p, nil
 }
+
+func (p *gateway) monitorChannelLen() {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for range ticker.C {
```

Comment:
We've got a worker type and this `gateway` type is composite with the `Worker` type:
https://github.com/katzenpost/katzenpost/blob/main/server/internal/gateway/gateway.go#L38
https://github.com/katzenpost/katzenpost/blob/main/core/worker/worker.go#L23

And so... a very common pattern we use inside the mix server is to maintain a server shutdown code path:

```golang
select {
        case <-p.HaltCh():
                return
        case <-timer.C:
                // blah blah
}
```

I'm not saying we need to do this here necessarily unless the mix server no longer shutdown down cleanly.